scene-lib/src/scenelib/annotations/el/LocalLocation.java

```diff
@@ -1,91 +1,217 @@
      */
-    public final int scopeStart;
+    public final int[] index;
 
     /**
-     * The length of the local variable's scope (or live range), in bytes.
+     * The name of the local variable being visited.
      */
-    public final int scopeLength;
+    public final String variableName;
 
-    public final String varName;
-    public final int varIndex;
+    // We now use the public ASM bytecode framework instead of our own local copy.
+    // As part of the implementation changes, the following member variables
+    // of LocalLocation no longer exist:
+    //   int scopeStart;
+    //   int scopeLength;
+    //   int varIndex;
+    // Instead, the following accessor methods must be used:
+    //   int getScopeStart();
+    //   int getScopeLength();
+    //   int getVarIndex();
 
     /**
-     * Constructs a new {@link LocalLocation}; the arguments are assigned to
-     * the fields of the same names.
+     * Construct a new LocalLocation.
+     *
+     * @param variableName the name of the local variable
+     * @param index the offset of the variable in the stack frame
      */
-    public LocalLocation(int index, int scopeStart, int scopeLength) {
+    public LocalLocation(String variableName, int index) {
+        this(new Label[] {new Label()}, new Label[] {new Label()}, new int[] {index}, variableName);
+    }
+
+    /**
+     * Construct a new LocalLocation.
+     *
+     * @param start an array of Labels indicating the start of the variable's lifetime(s)
+     * @param end an array of Labels indicating the end of the variable's lifetime(s)
+     * @param index an array of ints indicating the stack offset of the variable's lifetime(s)
+     * @param variableName the name of the local variable
+     */
+    public LocalLocation(Label[] start, Label[] end, int[] index, String variableName) {
+        this.start = start;
+        this.end = end;
         this.index = index;
-        this.scopeStart = scopeStart;
-        this.scopeLength = scopeLength;
-        this.varName = null;
-        this.varIndex = -1;
+        this.variableName = variableName;
     }
 
-    public LocalLocation(String varName, int varIndex) {
-        this.index = -1;
-        this.scopeStart = -1;
-        this.scopeLength = -1;
-        this.varName = varName;
-        this.varIndex = varIndex;
+    /**
+     * Construct a new LocalLocation.
+     *
+     * @param index the offset of the variable in the stack frame
+     * @param scopeStart the bytecode offset of the start of the variable's lifetime
+     * @param scopeLength the bytecode length of the variable's lifetime
+     */
+    public LocalLocation(int index, int scopeStart, int scopeLength) {
+        // Only being used by Writers, not Readers for now. Should possibly deprecate this in the future.
+        // Changes values reflectively.
+        this.index = new int[] {index};
+        this.start = new Label[] {new Label()};
```

Comment:
I think this code would be clearer if:
 * You create two variables for startLabel and endLabel.
 * Use them in the code below, instead of `start[0]` and `end[0]`.
 * Move this code that sets the fields to the very end of the constructor.